sympy/core/tests/test_sympify.py

```diff
@@ -626,10 +626,11 @@ def equal(x, y):
     assert equal(sympify(np.float64(1.1234567891234)),
                 Float(1.1234567891234, precision=53))
     assert equal(sympify(np.longdouble(1.123456789)),
-                 Float(1.123456789, precision=80))
+                 Float(1.123456789, precision=64))
     assert equal(sympify(np.complex64(1 + 2j)), S(1.0 + 2.0*I))
     assert equal(sympify(np.complex128(1 + 2j)), S(1.0 + 2.0*I))
-    assert equal(sympify(np.longcomplex(1 + 2j)), S(1.0 + 2.0*I))
+    assert equal(sympify(np.longcomplex(1 + 2j)),
```

Comment:
Why does this become `precision=64`? Shouldn't it be 128 since a longcomplex is 256 bit https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.clongdouble

Reply:
The precision refers to the number of bits in the mantissa so e.g. np.float64 corresponds to a precision of 53 bits. I think np.float128 should have 113 bits of precision. That's not what sympify does though:
```python
In [29]: sympify(np.longcomplex(1 + 2j)).args[0]._prec
Out[29]: 64
```

Reply:
So then one could state that the result of `sympify(np.longcomplex(1 + 2j))` is incorrect? Since it creates a `Float` with the wrong precision?

Reply:
The code in `sympify` is using `np.finfo(dtype).nmant`, which does return `63` for `np.longcomplex` https://github.com/sympy/sympy/blob/42248dd7be300bc3893642fd92793c0ff4c19a89/sympy/core/sympify.py#L91

Reply:
It comes from here:
https://github.com/sympy/sympy/blob/42248dd7be300bc3893642fd92793c0ff4c19a89/sympy/core/sympify.py#L90-L96
We can try that manually:
```python
In [7]: np.finfo(np.float128(1)).nmant + 1
Out[7]: 64

In [8]: np.finfo(np.float64(1)).nmant + 1
Out[8]: 53
```
The numpy docs say 
```
nmant : int
The number of bits in the mantissa.
```
https://numpy.org/doc/stable/reference/generated/numpy.finfo.html

However if you look here then you can see that the actual size of these datatypes is platform dependent:
https://numpy.org/doc/stable/user/basics.types.html#extended-precision

So actually np.float128 is most likely not the same as IEEE 128 bit floating point. I guess that on my computer (and also in CI) what we actually get is an 80-bit floating point number (as was used in x87 FPUs in the 32 bit days):
https://en.wikipedia.org/wiki/Extended_precision#Extended_precision_implementations

Reply:
I wonder if `finfo` then is not accurate. Why would `sympy.sympify(np.complex64(1 + 2j)).args[0]._prec == sympy.sympify(np.complex128(1 + 2j)).args[0]._prec == 53`?

Reply:
I see, so then the test should probably be rewritten in some platform independent way?

Reply:
See https://stackoverflow.com/questions/9062562/what-is-the-internal-precision-of-numpy-float128/17023995#17023995 and https://github.com/numpy/numpy/issues/10288.

Reply:
I've changed the test to be based on np.finfo